test/display_tests/display.jl

```diff
@@ -8,235 +8,238 @@ let x, b
-    =#
 
     setprecision(BigFloat, 256)
 
     @testset "DecoratedInterval" begin
         a = @decorated(1, 2)
-        @test typeof(a)== DecoratedInterval{Float64}
+        @test typeof(a) == DecoratedInterval{Float64}
 
-        setformat(:standard, decorations=false)
-        @test string(a) == "[1, 2]"
+        setformat(:standard; decorations = false)
+        @test sprint(show, MIME("text/plain"), a) == "[1.0, 2.0]"
 
-        setformat(:standard, decorations=true)
-        @test string(a) == "[1, 2]_com"
+        setformat(; decorations = true)
+        @test sprint(show, MIME("text/plain"), a) == "[1.0, 2.0]_com"
 
         # issue 131:
         a = DecoratedInterval(big(2), big(3), com)
 
-        setformat(:standard, decorations=false)
-        @test string(a) == "[2, 3]₂₅₆"
+        setformat(; decorations = false)
+        @test sprint(show, MIME("text/plain"), a) == "[2.0, 3.0]₂₅₆"
 
-        setformat(decorations=true)
-        @test string(a) == "[2, 3]₂₅₆_com"
+        setformat(; decorations = true)
+        @test sprint(show, MIME("text/plain"), a) == "[2.0, 3.0]₂₅₆_com"
 
         setformat(:full)
-        @test string(a) == "DecoratedInterval(Interval(2.0, 3.0), com)"
+        @test sprint(show, MIME("text/plain"), a) == "DecoratedInterval(Interval(2.0, 3.0), com)"
 
         setformat(:midpoint)
-        @test string(a) == "2.5 ± 0.5_com"
-
-        setformat(decorations=false)
-        @test string(a) == "2.5 ± 0.5"
+        @test sprint(show, MIME("text/plain"), a) == "2.5 ± 0.5_com"
 
+        setformat(; decorations = false)
+        @test sprint(show, MIME("text/plain"), a) == "2.5 ± 0.5"
     end
 
     setprecision(BigFloat, 128)
 
     @testset "BigFloat intervals" begin
-        setformat(:standard, decorations=false)
-
         a = Interval(big(1))
         @test typeof(a) == Interval{BigFloat}
-        @test string(a) == "[1, 1]₁₂₈"
 
-        setformat(:full)
-        @test string(a) == "Interval(1.0, 1.0)"
+        setformat(:standard; decorations = false)
+        @test sprint(show, MIME("text/plain"), a) == "[1.0, 1.0]₁₂₈"
 
+        setformat(:full)
+        @test sprint(show, MIME("text/plain"), a) == "Interval(1.0, 1.0)"
 
         a = DecoratedInterval(big(2), big(3), com)
-        @test typeof(a)== DecoratedInterval{BigFloat}
+        @test typeof(a) == DecoratedInterval{BigFloat}
 
-        setformat(:standard, decorations=false)
-        @test string(a) == "[2, 3]₁₂₈"
+        setformat(:standard; decorations = false)
+        @test sprint(show, MIME("text/plain"), a) == "[2.0, 3.0]₁₂₈"
 
-        setformat(:standard, decorations=true)
-        @test string(a) == "[2, 3]₁₂₈_com"
+        setformat(:standard; decorations = true)
+        @test sprint(show, MIME("text/plain"), a) == "[2.0, 3.0]₁₂₈_com"
 
         setformat(:full)
-        @test string(a) == "DecoratedInterval(Interval(2.0, 3.0), com)"
+        @test sprint(show, MIME("text/plain"), a) == "DecoratedInterval(Interval(2.0, 3.0), com)"
```

Comment:
This is a comment/suggestion, which I guess is small enough to be incorporated for intervals/decorated intervals with `BigFloat`s: I like that the `:standard` format shows the precision of the BigFloat end-points; this is not included in the `:full` format, as shown by this example. I think displaying the precision is informative; while we *usually* fix it (and forget it), MPFR allows to work with different precisions, which is a case where it is worth a reminder. 
So, what about including the precision for `Interval{BigFloat}` and `DecoratedInterval{BigFloat}` always?

Reply:
It is not a bad idea. Note that we need a signature for `show` which returns parseable Julia code (cf. docstring for `show`). In this PR this behaviour is tied with the formatting mechanism via `setformat(:full)`.
I think the best option would be to add an other symbol, say `:raw` (or perhaps `:parseable` 🤷‍♂️ ), such that `setformat(:raw)` replaces the current`setformat(:full)`. Then, `setformat(:full)` can have the behaviour you suggest.

Also, do you have a display in mind? Something like `@biginterval(1)` is displayed as `"Interval(1.0, 1.0)₂₅₆"`?

Reply:
For consistency, what about doing this for `:midpoint` as well?

So it would look like
```julia
julia> setformat(:midpoint)
Display options:
  - format: midpoint
  - decorations: true
  - significant digits: 6

julia> big(1)..2
(1.5 ± 0.5)₂₅₆

julia> setformat(:full);  # full display of the bounds and show precision if BigFloat

julia> big(1)..2
Interval(1.0, 2.0)₂₅₆

julia> setformat(:parseable);  # parseable Julia code

julia> big(1)..2
Interval(1.0, 2.0)
```

Reply:
Great that the comment was useful! 😄 

I like all proposals:
- the `:raw` and `:full` formats, where the raw format is for copy-paste (or saving results to be reused), and the full format is to have all details displayed;
- I like the display you propose `Interval(1.0, 1.0)₂₅₆`;
- include the precision in the `:midpoint` format.

Reply:
But now I am not unsure about the `:raw`/`:parseable` options 🤔. Mainly because the way Julia gives information of the precision for `BigFloat` is through the function `precision` and not the display.
So showing `Interval{BigFloat}` is sufficient and consistent with Base. A user would know that they have an interval with bounds of type `BigFloat` which has precision `precision(BigFloat)`.

But it is definitely useful to display the precision for `:midpoint`. So I will add a commit for this.